src/main/java/org/springframework/cli/merger/ProjectMergerWithMavenDepRecipe.java

```diff
@@ -114,32 +109,26 @@ public void merge() throws IOException {
 			AddManagedDependency addManagedDependency = getRecipeAddManagedDependency(dependency.getGroupId(), dependency.getArtifactId(), dependency.getVersion(), dependency.getScope(),
 					dependency.getType(), dependency.getClassifier());
 
-			List<? extends SourceFile> pomFiles = mavenParser.parse(paths, this.pathCurrentProject, getExecutionContext());
-			List<Result> resultList = addManagedDependency.run(pomFiles).getResults();
+			List<SourceFile> pomFiles = mavenParser.parse(paths, this.pathCurrentProject, getExecutionContext()).toList();
+			List<Result> resultList = addManagedDependency.run(new InMemoryLargeSourceSet(pomFiles), getExecutionContext()).getChangeset().getAllResults();
 			updatePomFile(currentProjectPomPath, resultList);
 		}
 
 		// Dependencies Section
 
 
-		/**
-		 * this approach does not work in our use case since it needs to construct a classpath for the application, that is the 'spring up shell'.
-		 * The rationale is that the recpie does not want to add a dependency, in particular in a multi-module project, to module that does not
-		 * use the code.  This involves loading classes from the classpath.  fall back to simple XML manipulation.
-		 */
-
-
 
 		List<Dependency> dependenciesToMerge = modelToMerge.getDependencies();
 
 		// Need to get source files
-		// TODO detect if java 11 or 8 or whatever....? JavaParser.fromJavaVersion() ?
-		JavaParser javaParser = new Java17Parser.Builder()
+		JavaParser javaParser = JavaParser.fromJavaVersion()
```

Comment:
@jvalkeal Does changing to dynamic lookup of JavaParser impact success of cretaing native app?

Reply:
dunno, we'll see when at some point graal build get more stable.